backend/src/main/java/com/timlenny/backend/service/TopicService.java

```diff
@@ -40,12 +40,36 @@ public Topic addTopic(TopicDTO topicToAddDTO) {
         }
 
         Optional<Topic> parentTopic = topicRepository.findByTopicName(topicToAddDTO.getParentName());
-        if (parentTopic.isPresent()) {
-            Topic topicToAdd = topicConversionService.convertNewDTOtoTopic(topicToAddDTO, parentTopic.get());
-            topicRepository.save(topicToAdd);
-            return topicToAdd;
-        } else {
-            throw new ResponseStatusException(HttpStatus.NO_CONTENT, "parentTopic not found!");
+
+        if (parentTopic.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "parentTopic not found!");
+        }
+
+        Topic topicToAdd = topicConversionService.convertNewDTOtoTopic(topicToAddDTO, parentTopic.get());
+        topicRepository.save(topicToAdd);
+        return topicToAdd;
+    }
+
+    public String deleteTopic(String id) {
+            List<String> deleteIds = new ArrayList<>();
+            List<String> resultDeleteIds = getAllChildTopics(id, deleteIds);
+            resultDeleteIds.add(id);
+
+            for (String delId : resultDeleteIds) {
+                topicRepository.deleteById(delId);
+            }
+
+            return id;
+    }
+
+    private List<String> getAllChildTopics(String parentId, List<String> deleteIds) {
+        List<Topic> childTopics = topicRepository.findByParentId(parentId);
+
+        for (Topic child : childTopics) {
+            deleteIds.add(child.getId());
+            getAllChildTopics(child.getId(), deleteIds);
         }
+
+        return deleteIds;
```

Comment:
Attention: this is a bit ambiguous. You are doing two things at the same time: You are mutating the passed `deleteIds` List but also returning a List at the end.
From a method I would expect to do either one or the other.
You could avoid this by declaring another `getAllChildTopics(String parentId)` - which doesn't get passed the List but creates it, then calls the method you already have and finally returns the final list. From the method you already have, you could remove the return statement. Maybe that makes more sense, but maybe that is as unclear to other people as this was to me ;)